os/pm/pm_metrics.c

```diff
@@ -17,117 +17,297 @@
  ****************************************************************************/
 
 #include <tinyara/config.h>
+#include <tinyara/pm/pm.h>
 #include <time.h>
 #include <queue.h>
 #include <debug.h>
-#include "pm_metrics.h"
+#include <errno.h>
 #include "pm.h"
 
-time_t time_diff(time_t time1, time_t time2)
+struct pm_metric_domain_s {
+	clock_t stime[CONFIG_PM_NDOMAINS];                           /* Last suspended time stamp of domain */
+	uint32_t blocking_board_sleep_counts[CONFIG_PM_NDOMAINS];    /* It counts the suspended domain inside idle thread which prevent board sleep */
+	uint32_t suspend_ticks[CONFIG_PM_NDOMAINS];                  /* Time (in ticks) domain suspended */
+};
+
+typedef struct pm_metric_domain_s pm_metric_domain_t;
+
+struct pm_metric_state_s {
+	clock_t stime;                                               /* Last PM change state time stamp */
+	uint32_t state_accum_ticks[PM_COUNT];                        /* PM State Time (in ticks)*/
+};
+
+typedef struct pm_metric_state_s pm_metric_state_t;
+
+struct pm_metric_s {
+	pm_metric_domain_t domain_metrics;                          /* The domain metrics */
+	pm_metric_state_t state_metrics;                            /* The power management state metrics */
+	uint32_t board_sleep_ticks;                                 /* The amount of time (in ticks) board was in sleep */
+	uint32_t wakeup_src_counts[PM_WAKEUP_SRC_COUNT];            /* It counts the frequency of wakeup sources */
+	uint32_t total_try_counts;                                  /* It counts no. of pm tries to make board sleep */
```

Comment:
total try count, is uint32_t enough?

Reply:
Assuming 1 count is equivalent to 1ms.

UINT_MAX_32 count is equivalent to 49.7 days. I think it's enough

Reply:
thank you for confirm 
the delay of wd_start only supports the integer type max value.
the pm_sleep using wdog is the same.

Therefore, it would be enough if pm_metrics is also designed to support ticks up to integer max.